src/Simplex/Messaging/Agent.hs

```diff
@@ -147,44 +150,121 @@ processCommand AgentClient {sndQ, msgQ, smpClients} (corrId, connAlias, cmd) =
               status = New,
               ackMode = AckMode On
             }
-      liftSMP $ sendSMPMessage c "" senderId msg
+      liftSMP $ sendSMPMessage smp "" senderId msg
       withStore $ \st -> updateQueueStatus st connAlias SND Confirmed
+      sendHello smp encryptKey sndPrivateKey senderId
+      withStore $ \st -> updateQueueStatus st connAlias SND Active
+      -- qInfo <- createReplyQueue replySrv
+      -- sendReplyQueue qInfo
       respond OK
 
-    replyError :: ErrorType -> SomeException -> m a
-    replyError err e = do
-      liftIO . putStrLn $ "Exception: " ++ show e -- TODO remove
-      throwError err
+    mkConfirmation :: PublicKey -> PublicKey -> m SMP.MsgBody
```

Comment:
Should we create separate types for different keys? I.e. EncryptionPublicKey, SenderPublicKey.
This may just be an improvement further down the line, not in this PR